### PR TITLE
cov: add tests for ignite.metrics

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1790 

Description: Add tests / Removed unused code for ignite.metrics

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/metric.py [hunk 1]
```diff
@@ -478,16 +478,6 @@ def __mod__(self, other: Any) -> "MetricsLambda":
 
         return MetricsLambda(lambda x, y: x % y, self, other)
 
-    def __div__(self, other: Any) -> "MetricsLambda":
```
Comment (ydcjeff) [inline ignite/metrics/metric.py:481]: These are only applicable to py2

File: ignite/metrics/rouge.py [hunk 2]
```diff
@@ -185,8 +185,6 @@ def _get_multiref_reducer(self) -> MultiRefReducer:
             return MultiRefAverageReducer()
         elif self._multiref == "best":
             return MultiRefBestReducer()
-        else:
-            raise ValueError(f"multiref : wrong value (got : {self._multiref})")
```
Comment (ydcjeff) [inline ignite/metrics/rouge.py:189]: This is no need to catch. The same error has been caught above.

**Comment by vfdev-5** [comment]
@ydcjeff, just a small though on saving computational resources while merging master to all PRs. When PR is approved we can update it once and then merge and avoid updating all PRs.

File: tests/ignite/metrics/test_metric.py [hunk 3]
```diff
@@ -220,24 +228,6 @@ def compute(self):
     m2.update([1, 10, 100])
     assert m2_mod_2.compute() == 0
 
-    # __div__, only applicable to python2
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_metric.py:223]: Let's add however tests like 
```python
# division
    m0_div_m1 = m0 / m1
    m0.update([1, 10, 100])
    m1.update([1, 10, 100])
    assert m0_div_m1.compute() == 0.1
    m1.update([2, 20, 200])
    m0.update([2, 20, 200])
assert m0_div_m1.compute() == 10
```

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @ydcjeff !

**Pull request merged**